src/main/java/com/labsynch/labseer/service/BulkLoadServiceImpl.java

```diff
@@ -381,159 +382,16 @@ public BulkLoadRegisterSDFResponseDTO registerSdf(BulkLoadRegisterSDFRequestDTO
-					continue;
-				}
-
-				Metalot metalot = new Metalot();
-				metalot.setLot(lot);
-				metalot.setIsosalts(lot.getSaltForm().getIsoSalts());
-				metalot.setSkipParentDupeCheck(true);
-				MetalotReturn metalotReturn = null;
-
-				if (validate) {
-					try {
-						parent = validateParentAgainstDryRunCompound(parent, numRecordsRead, results);
-						// Check list of aliases within file being bulkloaded
-						if (!propertiesUtilService.getAllowDuplicateParentAliases()) {
-							for (ParentAlias alias : parent.getParentAliases()) {
-								if (allAliasMaps.get(alias.getAliasName()) == null) {
-									allAliasMaps.put(alias.getAliasName(), numRecordsRead);
-								} else {
-									throw new NonUniqueAliasException(
-											"Within File, Parent Alias " + alias.getAliasName()
-													+ " is not unique");
-								}
-							}
-						}
-						saveDryRunCompound(mol, parent, numRecordsRead, dryRunCompound);
-
-					} catch (TransactionSystemException rollbackException) {
-						logger.error("Rollback exception", rollbackException.getApplicationException());
-						Exception causeException = new Exception(
-								rollbackException.getApplicationException().getMessage(),
-								rollbackException.getApplicationException());
-						logError(causeException, numRecordsRead, mol, mappings, errorMolExporter, results,
-								errorCSVOutStream);
-						continue;
-					} catch (Exception e) {
-						logError(e, numRecordsRead, mol, mappings, errorMolExporter, results, errorCSVOutStream);
-						continue;
-					}
-
-				}
-				if (!validate) {
-					try {
-						metalotReturn = metalotService.save(metalot);
-						if (!metalotReturn.getErrors().isEmpty()) {
-							for (ErrorMessage errorMessage : metalotReturn.getErrors()) {
-								Exception e = new Exception(errorMessage.getMessage());
-								logError(e, numRecordsRead, mol, mappings, errorMolExporter, results,
-										errorCSVOutStream);
-							}
-							continue;
-						}
-					} catch (Exception e) {
-						logError(e, numRecordsRead, mol, mappings, errorMolExporter, results, errorCSVOutStream);
-						continue;
+				try{
+					Boolean isNewParent = registerMol(mol, bulkLoadFile, mappings, errorCSVOutStream, registeredCSVOutStream, errorMolExporter, registeredMolExporter, allAliasMaps, numRecordsRead, results, validate, chemist, registerRequestDTO, dryRunCompound, startTime);
+					if(isNewParent != null) {
+						if (isNewParent)
+							numNewParentsLoaded++;
+						else
+							numNewLotsOldParentsLoaded++;
 					}
-				}
-
-				// Successful registration/validation: update summary and write
-				// registered/validated record
-				if (isNewParent)
-					numNewParentsLoaded++;
-				else
-					numNewLotsOldParentsLoaded++;
-				writeRegisteredMol(numRecordsRead, mol, metalotReturn, mappings, registeredMolExporter,
-						registeredCSVOutStream, isNewParent, results);
-				if (numRecordsRead % 100 == 0) {
-					logger.info("flushing bulk loader session");
```

Comment:
I think you can also remove the Session on line 287/288 since we're no longer using it, and remove the import

Reply:
Thanks, I removed it.